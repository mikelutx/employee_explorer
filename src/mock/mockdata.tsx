import { mEmployee } from "./../model/mEmployee";

const user1: mEmployee = {
  name: "John Hartman",
  title: "CEO",
  subordinates: ["Samad Pitt", "Leanna Hogg"],
};

const user2: mEmployee = {
  name: "Samad Pitt",
  title: "production supervisor",
  subordinates: ["Aila Hodgson", "Amaya Knight"],
};

const user3: mEmployee = {
  name: "Leanna Hogg",
  title: "marketing supervisor",
  subordinates: ["Vincent Todd", "Faye Oneill", "Lynn Haigh", "Aila Hodgson"],
};

export const mockEmployeeSearchResult: mEmployee[] = [user1, user2];

export const mockEmployee: mEmployee = user1;

export const mockEmployeeSubs: mEmployee[] = [user2, user3];
