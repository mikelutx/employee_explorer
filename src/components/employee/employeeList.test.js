import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { mockEmployeeSearchResult } from "../../mock/mockdata";
import { act } from "react-dom/test-utils";
import pretty from "pretty";
import EmployeeList from "./employeeList";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
  act(() => {
    render(<EmployeeList employeeList={mockEmployeeSearchResult} />, container);
  });
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

test("total number of employee matches", () => {
  const cards = document.querySelectorAll(".card");
  expect(cards.length).toBe(mockEmployeeSearchResult.length);
});

test("1st employee name matches", () => {
  const title = document.querySelector(".card-title");
  expect(title?.innerHTML).toBe(mockEmployeeSearchResult[0].name);
});

test("snapshot testing employee list", () => {
  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
