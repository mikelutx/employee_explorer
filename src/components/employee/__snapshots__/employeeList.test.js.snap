// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing employee list 1`] = `
"<div class=\\"row\\">
  <div class=\\"col-sm-6 col-md-4 col-lg-3 mb-3\\">
    <div class=\\"card\\">
      <div class=\\"card-body\\">
        <h5 class=\\"card-title\\">John Hartman</h5>
        <h6 class=\\"card-subtitle mb-2 text-muted\\">CEO</h6>
        <p>2 subordinate(s)</p><a href=\\"/\\" class=\\"btn btn-primary\\">View</a>
      </div>
    </div>
  </div>
  <div class=\\"col-sm-6 col-md-4 col-lg-3 mb-3\\">
    <div class=\\"card\\">
      <div class=\\"card-body\\">
        <h5 class=\\"card-title\\">Samad Pitt</h5>
        <h6 class=\\"card-subtitle mb-2 text-muted\\">production supervisor</h6>
        <p>2 subordinate(s)</p><a href=\\"/\\" class=\\"btn btn-primary\\">View</a>
      </div>
    </div>
  </div>
</div>"
`;
